relay-server/src/utils/rate_limits.rs

```diff
@@ -709,52 +863,120 @@ where
             rate_limits.merge(event_limits);
         }
 
+        // Handle spans.
+        if enforcement.is_event_active() {
+            enforcement.spans = enforcement
+                .event
+                .clone_for(DataCategory::Span, summary.span_quantity);
+
+            enforcement.spans_indexed = enforcement
+                .event_indexed
+                .clone_for(DataCategory::SpanIndexed, summary.span_quantity);
+        // Check the span quota if either there are spans or there are span attachments. Note: in
+        // the later case span_quantity is 0 so no quota will be consumed by the check.
+        } else if summary.span_quantity > 0 || !summary.attachment_quantities.span.is_empty() {
+            let mut span_limits = self
+                .check
+                .apply(scoping.item(DataCategory::Span), summary.span_quantity)
+                .await?;
+            enforcement.spans = CategoryLimit::new(
+                DataCategory::Span,
+                summary.span_quantity,
+                span_limits.longest(),
+            );
+
+            if span_limits.is_empty() {
+                span_limits.merge(
+                    self.check
+                        .apply(
+                            scoping.item(DataCategory::SpanIndexed),
+                            summary.span_quantity,
+                        )
+                        .await?,
+                );
+            }
+
+            enforcement.spans_indexed = CategoryLimit::new(
+                DataCategory::SpanIndexed,
+                summary.span_quantity,
+                span_limits.longest(),
+            );
+
+            rate_limits.merge(span_limits);
+        }
+
+        // Handle span attachments (part 1)
+        if !summary.attachment_quantities.span.is_empty()
+            && (!enforcement.spans.is_default() || !enforcement.spans_indexed.is_default())
+        {
+            let span_limit = if !enforcement.spans.is_default() {
+                &enforcement.spans
+            } else {
+                &enforcement.spans_indexed
+            };
+            enforcement.attachments_limits.span.bytes = span_limit.clone_for_unchecked(
```

Comment:
Discussed offline, the `span_indexed` limit is enough as it already inherits the total limit.